zaza/charm_tests/security/tests.py

```diff
@@ -57,7 +46,9 @@ def class_decorator(cls):
             # Lets make sure to only add tests for deployed applications
             if name in deployed_applications:
                 for file in attributes['files']:
-                    test_func = _make_test_function(name, file)
+                    paths = [file['path'] for file in attributes['files']]
+                    paths = [path for path in paths if path[-1] is not "*"]
```

Comment:
Can the above two dict comprehensions be moved above the enclosing loop ? Having them in the same loop and reusing the same variable name to iterate over the dict is a bit confusing.